app/src/features/rules/screens/sharedListViewer/components/SharedListViewerList/SharedListViewerList.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { useState } from "react";
+import { useSelector } from "react-redux";
+import { getIsAppBannerVisible } from "store/selectors";
+import { Empty } from "antd";
+import { ContentListTable } from "componentsV2/ContentList";
+import { RuleTableRecord } from "features/rules/screens/rulesList/components/RulesList/components/RulesTable/types";
+import { useSharedListViewerColumns } from "./hooks/useSharedListViewerColumns";
+import { SharedListViewerModal } from "../../modals/SharedListRuleViewerModal";
+import "./sharedListViewerList.scss";
+
+interface Props {
+  records: RuleTableRecord[];
+  isLoading: boolean;
+}
+
+export const SharedListViewerList: React.FC<Props> = ({ records, isLoading }) => {
+  const [isSharedListViewerModalVisible, setIsSharedListViewerModalVisible] = useState(false);
+  const [ruleToView, setRuleToView] = useState(null);
+  const isAppBannerVisible = useSelector(getIsAppBannerVisible);
+
+  const handleViewSharedListRule = (rule: RuleTableRecord) => {
+    setRuleToView(rule);
+    setIsSharedListViewerModalVisible(true);
+  };
+  const columns = useSharedListViewerColumns({ handleViewSharedListRule });
+
+  return (
+    <>
+      <ContentListTable
+        columns={columns}
+        data={records}
+        rowKey="id"
+        id="shared-list-viewer-table"
+        className="shared-list-viewer-table"
+        locale={{
+          emptyText: <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} description="No Shared list found" />,
+        }}
+        loading={isLoading}
+        scroll={isAppBannerVisible ? { y: "calc(100vh - 232px - 48px)" } : undefined}
```

Comment:
Is `undefined` a valid value for `scroll` prop, please check confirm it once

Reply:
I am passing undefined here so that it takes the default scroll value defined in the component. Passing null override it as well.